pydantic/_internal/_std_types_schema.py

```diff
@@ -95,6 +95,12 @@ def get_json_schema(_, handler: GetJsonSchemaHandler) -> JsonSchemaValue:
         update_json_schema(original_schema, updates)
         return json_schema
 
+    strict_python_schema = core_schema.is_instance_schema(enum_type)
+    if use_enum_values:
+        strict_python_schema = core_schema.chain_schema(
+            [strict_python_schema, core_schema.no_info_plain_validator_function(lambda x: x.value)]
```

Comment:
I think what happens is that, when parsing unions, it tries to parse in strict mode first so that it gets the right union case, rather than the first one, when possible. Because the strict mode python schema wasn't applying the `use_enum_values` thing before, it didn't affect the parsing of unions.

I guess it was also a bug that if you were to validate an enum in strict mode with `use_enum_values=True` it wouldn't actually use the enum values either.